tool/speed.cc

```diff
@@ -1161,6 +1161,25 @@ static bool SpeedHmac(const EVP_MD *md, const std::string &name,
   if (!selected.empty() && name.find(selected) == std::string::npos) {
     return true;
   }
+  TimeResults results;
+  const size_t key_len = EVP_MD_size(md);
+  std::unique_ptr<uint8_t[]> key(new uint8_t[key_len]);
+  BM_memset(key.get(), 0, key_len);
+
+  if (!TimeFunction(&results, [&]() -> bool {
+#if defined(OPENSSL_1_0_BENCHMARK)
+        BM_NAMESPACE::UniquePtr<HMAC_CTX> ctx(new HMAC_CTX);
+        HMAC_CTX_init(ctx.get());
+#else
+        BM_NAMESPACE::UniquePtr<HMAC_CTX> ctx(HMAC_CTX_new());
```

Comment:
just to confirm -- we want to include the time to allocate a new HMAC_CTX here, right?

also, why do we call both `HMAC_CTX_init` and `HMAC_Init_ex` in the case of OpenSSL 1.0?

Reply:
That's a good question, I was just following what we do in the overall HMAC benchmark https://github.com/aws/aws-lc/blob/main/tool/speed.cc#L1120-L1128. [HMAC_CTX_init](https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/hmac/hmac.c#L202-L208) init's the MD_CTX's, then [HMAC_Init_ex](https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/hmac/hmac.c#L68-L156) sets the key. 

Looking at 1.1+ and AWS-LC's HMAC_CTX_new it's the same as this 1.0 code path: malloc a `HMAC_CTX` then call `HMAC_CTX_init`.

Reply:
But to your point I'm leaning towards just measuring`HMAC_Init_ex` now that you mention it.

Reply:
One doesn't actually have to `HMAC_CTX_new()` in AWS-LC. heap + an `HMAC_CTX_init()` is enough because the object is completely flat. Was done to improve memory management overhead from ACCP.

Reply:
@htorben that sounds about right. ACCP currently [allocates an array on the java heap](https://github.com/corretto/amazon-corretto-crypto-provider/blob/ec7e08e2cafcbc78bbe47ba5f1542304e56f2e89/src/com/amazon/corretto/crypto/provider/EvpHmac.java#L204) for the context object, then [relies on AWS-LC to initialize that array](https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/csrc/hmac.cpp#L23-L37).